Sprint-2/implement/querystring.js

```diff
@@ -1,13 +1,24 @@
 function parseQueryString(queryString) {
   const queryParams = {};
-  if (queryString.length === 0) {
-    return queryParams;
-  }
+
+  if (!queryString) return queryParams;
+
   const keyValuePairs = queryString.split("&");
 
   for (const pair of keyValuePairs) {
-    const [key, value] = pair.split("=");
-    queryParams[key] = value;
+    const parts = pair.split("=");
+    const key = decodeURIComponent(parts[0]);
+    const value =
```

Comment:
Rather than splitting and rejoining, how might you do this with indexOf and slice?   Your method works, I just want you to consider alternatives.  In what circumstances is each approach better?  (you don't need to respond to that question here, just have a think about it - although feel free to do so if you want feedback on your answer)